### PR TITLE
Feature/before dev

### PR DESCRIPTION
**Pull request by Codefeak**
uses of both crowd or local according to availability

### CODE REVIEW

File: src/app.js [hunk 1]
```diff
@@ -93,22 +95,28 @@ export default async function run() {
...
+	app.use('/users', passportMiddlewares.token, createUsersRouter(db, combineUserInfo));
+	app.use('/requests/users', passportMiddlewares.token, createRequestsUsersRouter(db, combineUserInfo));
+	app.use('/publishers', createPublishersRouter(db, passportMiddlewares, combineUserInfo));
+	app.use('/requests/publishers', passportMiddlewares.token, createPublishersRequestsRouter(db, combineUserInfo));
+	app.use('/publications/isbn-ismn', passportMiddlewares.token, createPublicationsRouterIsbnIsmn(db, combineUserInfo));
+	app.use('/requests/publications/isbn-ismn', passportMiddlewares.token, createRequestsPublicationsRouterIsbnIsmn(db, combineUserInfo));
+	app.use('/publications/issn', passportMiddlewares.token, createPublicationsRouterIssn(db, combineUserInfo));
+	app.use('/requests/publications/issn', passportMiddlewares.token, createRequestsPublicationsRouterIssn(db, combineUserInfo));
+	app.use('/ranges', passportMiddlewares.token, createRangesRouter(db, combineUserInfo));
 	app.use('/auth', authenticationRouter(db, passportMiddlewares));
```
Comment (natlibfi-arlehiko) [inline src/app.js]: Why pass `combineUserInfo`? You can define this in a module all the routers can access and then import from there.

File: src/app.js [hunk 2]
```diff
@@ -58,14 +60,14 @@ const {Crowd: {generatePassportMiddlewares}} = Authentication;
...
+	const client = new MongoClient(MONGO_URI, {useNewUrlParser: true, useUnifiedTopology: true});
 	const connection = await client.connect();
 	const db = connection.db();
 
 	const passportMiddlewares = await generatePassportMiddlewares({
 		crowd: {
-			// AppName: CROWD_APP_NAME, appPassword: CROWD_APP_PASSWORD,
-			// url: CROWD_URL, useCache: true, fetchGroupMembership: true
+			appName: CROWD_APP_NAME, appPassword: CROWD_APP_PASSWORD,
+			url: CROWD_URL ? `${CROWD_URL}/rest` : null, useCache: true, fetchGroupMembership: true
```
Comment (natlibfi-arlehiko) [inline src/app.js]: `CROWD_URL` is never optional.

File: src/config.js [hunk 3]
```diff
@@ -12,6 +12,16 @@ export const API_CLIENT_USER_AGENT = readEnvironmentVariable('API_CLIENT_USER_AG
...
+export const CROWD_URL = readEnvironmentVariable('CROWD_URL', {
+	defaultValue: null
+});
+export const CROWD_APP_NAME = readEnvironmentVariable('CROWD_APP_NAME', {
+	defaultValue: null
+});
+export const CROWD_APP_PASSWORD = readEnvironmentVariable('CROWD_APP_PASSWORD', {
+	defaultValue: null
+});
+
```
Comment (natlibfi-arlehiko) [inline src/config.js]: `null` shouldn't be used unless explicitly needed. Change to `{defaultValue: ''}`

File: src/config.js [hunk 4]
```diff
@@ -29,8 +39,12 @@ export const HTTP_PORT = readEnvironmentVariable('HTTP_PORT', {
 export const MONGO_URI = readEnvironmentVariable('MONGO_URI', {
 	defaultValue: 'mongodb://localhost:27017/db'
 });
-export const PASSPORT_LOCAL_USERS = readEnvironmentVariable('PASSPORT_LOCAL_USERS');
-export const PASSPORT_LOCAL = readEnvironmentVariable('PASSPORT_LOCAL');
+export const PASSPORT_LOCAL_USERS = readEnvironmentVariable('PASSPORT_LOCAL_USERS', {
+	defaultValue: null
+});
+export const PASSPORT_LOCAL = readEnvironmentVariable('PASSPORT_LOCAL', {
```
Comment (natlibfi-arlehiko) [inline src/config.js]: Why this is needed when we have `PASSPORT_LOCAL_USERS`?

File: src/config.js (hunk 3 above)
Comment (Codefeak) [reply src/config.js]: changed default value

File: src/app.js (hunk 1 above)
Comment (Codefeak) [reply src/app.js]: I made the changes, take a peek if its alright

**Pull request merged**